src/libertem/common/buffers.py

```diff
@@ -62,17 +63,24 @@ def __init__(self, kind, extra_shape=(), dtype="float32"):
             The abstract shape of the buffer, corresponding either to the navigation
             or the signal dimensions of the dataset, or a single value.
 
-        extra_shape : optional, tuple of int
+        extra_shape : optional, tuple of int or a Shape object
             You can specify additional dimensions for your data. For example, if
             you want to store 2D coords, you would specify (2,) here.
+            For a Shape object, sig_dims is discarded and the entire shape is used.
 
         dtype : string or numpy dtype
             The dtype of this buffer
         """
+
+        if isinstance(extra_shape, Shape):
+            self._extra_shape = extra_shape._nav_shape + extra_shape._sig_shape
+        else:
+            self._extra_shape = extra_shape
```

Comment:
Even though it works, accessing internal attributes of other classes (starting with an underscore) should not be done - they are used to separate the interface from implementation details (in other languages, there are stricter mechanisms for ensuring this, like visibility specifiers `private` etc.). Imagine the `Shape` implementation is changed to store the whole shape as a single tuple (or even another type: list, ndarray?) - you would have to change `_nav_shape` and `_sig_shape` usages all over the code base.

There is an even better way to do the same: `Shape` implements `__iter__`, and  any iterable can be converted to a `tuple` using the `tuple()` built-in function. So there is also no need for the `isinstance` check anymore - just unconditionally call `tuple` on `extra_shape`.

That means we accept any iterable, not just tuple and `Shape`, [which is even better IMHO.](https://en.wikipedia.org/wiki/Robustness_principle)

Reply:
Thanks for the detailed explanation, I've updated accordingly